packages/blocks/callout/src/App.tsx

```diff
@@ -0,0 +1,54 @@
+import { BlockComponent } from "blockprotocol/react";
+import React, { CSSProperties, RefCallback } from "react";
+
+import { EmojiIcon } from "./EmojiIcon";
+
+type AppProps = {
+  editableRef?: RefCallback<HTMLElement>;
+  icon?: string;
+  text?: string;
+};
+
+const wrapperStyle: CSSProperties = {
+  borderRadius: "0.25em",
+  width: "100%",
+  position: "relative",
+  padding: "0.5em",
+  background: "#f9fafc",
+  border: "1px solid #dee7f3",
+};
+
+const textStyle: CSSProperties = {
+  minHeight: "1.5em",
+  paddingLeft: "1.5em",
+};
+
+export const App: BlockComponent<AppProps> = ({
+  editableRef,
+  icon = "📢",
+  text,
+  entityId,
+  updateEntities,
+}) => {
+  const handleIconChange = (newIcon: string | undefined): void => {
+    if (!entityId) {
```

Comment:
So currently when `entityId` is not set, you get a hover style on `EmojiIcon` indicating its clickable, but clicking does nothing. This isn't a great experience. @MaggieAppleton 

*(for context, `entityId` won't exist when you create a new callout until collab has saved the callout to the db for you, and delivered the new entityId to the frontend)*

Can we disable the hover style when `entityId` is not defined? Because clicking isn't going to work until it is, so it's just bit confusing when you click and nothing happens. `entityId`. This is the quick fix, but there are better things we can do later:

Another approach is that you optimistically update the view, and you trigger the update once entityId is present. 

Another option is we make HASH generate the entity ids for new blocks. Not sure if this is a security risk.

A final option is HASH gives blocks fake entity ids until collab gives us the real ones, caches any requested updates, and apply them once we know the real entity ids, and apply any updates made using the fake entity ids optimistically, so that blocks themselves don't have to worry about these implementation details

What do you think?

Reply:
I hope we'll be able to provide some `entityId` in all cases in the future, but in the meantime, I added disabled state based on `entityId` presence. In this mode, the icon is not clickable and the cursor is a default arrow, not a pointer.

I also inlined all CSS in the block for consistency (which addresses your other comment too).